here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/IntHandlerForStorages.java

```diff
@@ -57,4 +62,16 @@ public IntHandlerForStorages(final @NotNull INaksha hub) {
       return notImplemented(event);
     }
   }
+
+  private @NotNull Result validateWriteRequest(final @NotNull WriteRequest<?, ?> wr) {
+    Result result = null;
+    for (final WriteOp<?> wOp : wr.queries) {
+      final Storage storage = (Storage) wOp.feature;
```

Comment:
can this casting to (Storage) be removed, to prevent potential runtime error? Instead maybe have the type Storage in the parameter and the check on line 51, like WriteRequest<Storage, ?>, would that work?

Reply:
I am seconding that - it would great to utilize strict typing here

Reply:
What we get as input is Request , from there we obtain WriteRequest ... so, in my view it won't make difference whether we typecast to Storage upfront on WriteRequest itself or while accessing WriteRequest.queries() .. I am delaying the decision to typecast, till we reach to a point where we really need Storage, which is another way of looking at this problem. If you do it upfront, counter argument is also valid, as to why do it upfront, if we only need it in one particular function. In either ways, since we are already in Storage related handler, I don't see this to be a real problem.
Apart from this theory, Phuc tried the same practically in his EventHandler code and we together learnt that it is not making any difference.
Note - considering the design that we work with, due to inheritance of Request / Result objects and dynamic class loading, the typecasting at multiple places is expected and I would consider it as normal. Yes, we together need to be aware about bad (unexpected) typecasting and to be cautious about it.

Reply:
> I am delaying the decision to typecast, till we reach to a point where we really need Storage

This sentence + the fact that we will have to cast stuff at some point convinced me.
The real essence of this discussion lays in the note you made - we are doing dynamic typing in strictly typed language, we won't be able to avoid such cases (and it's not bad since it by design).